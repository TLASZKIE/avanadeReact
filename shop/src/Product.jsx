import React, { Component } from 'react'

class Product extends Component {
  constructor(props) {
    super(props)

    this.state = { toggle: false }
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick() {
    this.setState(prev => {
      return { toggle: !prev.toggle }
    })
  }

  render() {
    const { img, name, desc } = this.props
    return (
      <div>
        <img src={img} onClick={this.handleClick} />
        <div>{name}</div>
        {this.state.toggle ? <div>{desc}</div> : null}
      </div>
    )
  }
}

export default Product
