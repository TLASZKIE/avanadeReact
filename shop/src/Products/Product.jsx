import React from 'react'

export default function Product(props) {
  const { img, name, price, desc, specialOffer, toggle, onToggle } = props

  const specialClass = specialOffer ? 'special' : ''

  return (
    <div className={`products-product ${specialClass}`}>
      <img src={img} alt={name} onClick={onToggle} width="100" height="100" />
      <div>{name}</div>
      <div>{price}</div>
      {toggle ? <div>{desc}</div> : null}
    </div>
  )
}
