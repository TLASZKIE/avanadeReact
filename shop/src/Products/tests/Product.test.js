import React from 'react'
import ReactDOM from 'react-dom'
import Product from '../Product'
import renderer from 'react-test-renderer'

//4/ The most primitive smoke test.
it('renders without crashing', () => {
  const product = { name: 'name', id: 1, specialOffer: true, price: 1600 }

  const div = document.createElement('div')
  ReactDOM.render(<Product {...product} />, div)
})

it('renders correctly', () => {
  const product = { name: 'name', id: 1, specialOffer: true, price: 1600 }
  const tree = renderer.create(<Product {...product} />).toJSON()
  expect(tree).toMatchSnapshot()
})
