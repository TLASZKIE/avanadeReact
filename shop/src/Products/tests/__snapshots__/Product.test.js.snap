// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="products-product special"
>
  <img
    alt="name"
    height="100"
    onClick={undefined}
    src={undefined}
    width="100"
  />
  <div>
    name
  </div>
  <div>
    1600
  </div>
</div>
`;
